nominal/client.py

```diff
@@ -0,0 +1,481 @@
+import chdb
+from chdb import session as chs
+import polars as pl
+
+import pandas as pd
+import pyarrow as pa
+from uuid import uuid4
+import matplotlib.pyplot as plt
+
+import os
+import requests
+from requests.utils import CaseInsensitiveDict
+from conjure_python_client import ServiceConfiguration
+from functools import cache
+
+import keyring as kr
+
+from ._utils import create_service
+from .cloud import get_base_url
+
+
+TOKEN = kr.get_password("Nominal API", "python-client")
+
+# wrappers for conjure service apis
+
+from _api.scout_catalog import CatalogService
+
+catalog = create_service(CatalogService, get_base_url())
+
+from _api.timeseries_logicalseries import LogicalSeriesService
+
+logical_series = create_service(LogicalSeriesService, get_base_url())
+
+from _api.timeseries_seriescache import SeriesCacheService
+
+series_cache = create_service(SeriesCacheService, get_base_url())
+
+from _api.scout import RunService
+
+runs = create_service(RunService, get_base_url())
+
+
+def _create_dataset(filename: str, bucket: str, key: str, ts_name: str):
+    from _api.scout_catalog import (
+        TimestampType,
+        CreateDataset,
+        Handle,
+        S3Handle,
+        DatasetOriginMetadata,
+        TimestampMetadata,
+        AbsoluteTimestamp,
+        Iso8601Timestamp,
+    )
+
+    return catalog.create_or_update_dataset(
+        TOKEN,
+        CreateDataset(
+            name=filename,
+            handle=Handle(s3=S3Handle(bucket, key)),
+            labels=[],
+            metadata={},
+            origin_metadata=DatasetOriginMetadata(
+                timestamp_metadata=TimestampMetadata(
+                    series_name=ts_name,
+                    timestamp_type=TimestampType(absolute=AbsoluteTimestamp(iso8601=Iso8601Timestamp())),
+                )
+            ),
+            properties={},
+        ),
+    )
+
+
+from _api.timeseries_logicalseries_api import BatchCreateLogicalSeriesRequest, CreateLogicalSeries
+from _api.timeseries_seriescache_api import CachedSeries, CreateCachedSeriesRequest
+
+
+def _create_logical_series(requests: list[CreateLogicalSeries]):
+    return logical_series.batch_create_logical_series(TOKEN, BatchCreateLogicalSeriesRequest(requests=requests))
+
+
+def _create_run(name: str, datasets: list["Dataset"], start_time: int, end_time: int):
+    from _api.scout_run_api import CreateRunRequest, UtcTimestamp, DataSourceRefName, CreateRunDataSource, DataSource
+
+    data_sources = [
+        (ds.filename, CreateRunDataSource(data_source=DataSource(dataset=ds.dataset_rid), series_tags={}))
```

Comment:
will want the ability to create your own ref name here, with the filename probably as a fallback